src/components/atoms/UserAvatar/UserAvatar.scss

```diff
@@ -1,23 +1,34 @@
 @import "scss/constants";
 
 .UserAvatar {
-  $avatar-sizes-map: (
-    small: 25px,
-    medium: 40px,
-    large: 54px,
-    full: 100%,
-  );
-  $indicator-sizes-map: (
-    small: 8px,
-    medium: 10px,
-    large: 12px,
-    full: 25%,
-  );
-
-  // NOTE: parent and img mismatch due to legacy reasons
+  // @see https://math.stackexchange.com/questions/377234/circle-in-square-calculate-distance-from-squares-corner-to-circles-perimeter
+  $square-root-of-2: 1.41421356237;
+  $indicator-offset-factor: ($square-root-of-2 - 1) / 2;
+  $indicator-radius-factor: $square-root-of-2 / 2;
+
+  // indicator border should be 12% of the indicator size, but borders don't accept % so using px for default
+  $default-indicator-border-width: 2px;
+  $default-indicator-size: 28%;
   $default-avatar-size: 25px;
   $default-image-size: 100%;
-  $default-indicator-size: 8px;
+
+  // indicator border is 12% of indicator size which itself is some % of avatar size
+  // limited to min: 0.8px and max: 2px
+  @function indicator-border-width($indicator-size, $avatar-size) {
+    @return clamp(0.8px, 0.12 * #{$indicator-size} * #{$avatar-size}, 2px);
+  }
```

Comment:
Using function makes it easier to understand why some numbers get multiplied as well as a bit more robustness.